V1

orthagonal opened the pull request.

orthagonal committed:
v1

orthagonal committed:
add conf/default.yaml

--use env vars to set creds

jgallen23 commented on conf/default.yaml:
```diff
@@ -0,0 +1,8 @@
+  hapi-upload-s3:
+    endpoint: '/api/media'
+    s3Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
+    s3SecretAccessKey: '{{ENV.MEDIAMANAGER_SECRETKEY}}'
```
lets move the s3 stuff out into its own object and then reference them here.

jgallen23 commented on methods/get.js:
```diff
@@ -0,0 +1,12 @@
+const s3 = new AWS.S3({ apiVersion: '2006-03-01' });
+
+module.exports = function(key, callback) {
+  const options = this.registrations['hapi-upload-s3'].options;
```
switch this to use `this.settings.app.s3...`

jgallen23 commented on methods/get.js:
```diff
@@ -0,0 +1,12 @@
+const AWS = require('aws-sdk');
```
what does this method do?  return the image content?

orthagonal replied:
Yeah, it returns the raw image data.  If it was uploaded with an s3 key name ending in .jpg/.png/.etc, your browser will interpret it as a valid image file and download it.

jgallen23 commented on routes/test.js:
```diff
@@ -0,0 +1,21 @@
+  path: '/test',
+  method: 'GET',
+  handler: (request, reply) => {
+    reply(`
```
can we switch this to use vision with handlebars.  We'll need this later anyways

orthagonal committed:
use vision/handlebars,

refactors yaml

jgallen23 commented on methods/get.js:
```diff
@@ -2,11 +2,9 @@ const AWS = require('aws-sdk');
-  s3Options.Bucket = options.s3Bucket;
-  s3Options.Key = key;
-  s3.getObject(s3Options, (err, data) => {
+  const options = this.settings.app.s3;
```
you probably want to clone this, otherwise `Key` is set on the settings object

jgallen23 commented on conf/default.yaml:
```diff
@@ -1,8 +1,20 @@
+  Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
+  SecretAccessKey: '{{ENV.MEDIAMANAGER_SECRETKEY}}'
+  AccessKeyId: '{{ENV.MEDIAMANAGER_ACCESSKEY}}'
+  Region: 'us-east-1'
```
this should be an env var too

orthagonal replied:
Confi is actually throwing a yaml parse error when I do {{s3.region}} there, not sure why.  (fixed)

jgallen23 replied:
because it can't reference itself

jgallen23 commented on conf/default.yaml:
```diff
@@ -1,8 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.MEDIAMANAGER_BUCKET}}'
```
don't need `MEDIAMANAGER_` prefix

orthagonal committed:
use clone, fixes region to u

jgallen23 commented on conf/default.yaml:
```diff
@@ -0,0 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.BUCKET}}'
```
we typically don't capitalize key names

orthagonal replied:
These are reserved case-sensitive keynames used by aws-sdk. If we lowercase them in default.yaml we will still have to uppercase them when the method is called, before we can pass them to AWS.  I'm cool with it either way, let me know if you still want me to change this @jgallen23

jgallen23 commented on conf/default.yaml:
```diff
@@ -0,0 +1,20 @@
+debug: true
+s3:
+  Bucket: '{{ENV.BUCKET}}'
+  SecretAccessKey: '{{ENV.SECRETKEY}}'
```
we should use the aws versions. AWS_SECRET_KEY (I think)

jgallen23 commented on conf/default.yaml:
```diff
@@ -0,0 +1,20 @@
+    _priority: -1
+  hapi-upload-s3:
+    endpoint: '/api/media'
+    s3Bucket: '{{s3.bucket}}'
```
these won't work. don't match the key names

jgallen23 commented on conf/default.yaml:
```diff
@@ -0,0 +1,20 @@
+  hapi-upload-s3:
+    endpoint: '/api/media'
+    s3Bucket: '{{s3.bucket}}'
+    s3SecretAccessKey: '{{s3.secretaccesskey}}'
```
same as above, this is all lowercase, so won't work

jgallen23 commented:
make sure you are testing these things. I shouldn't be able to see errors by just looking at the code

orthagonal committed:
conf uses aws env vars, lowercase key names

--fixed methods.get to be shorter, clearer, and use lowercase key names
from conf
